informer/src/handler/rpc_ws_handler.rs

```diff
@@ -0,0 +1,103 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{
+    start_ws, Connection, Error, ErrorCode, Params, PubSubHandler, Rng, Session, Subscriber, SubscriptionId, Value,
+    WsError, WsServer,
+};
+use crossbeam::Sender;
+use crossbeam_channel as crossbeam;
+use futures::Future;
+use jsonrpc_core::{futures, BoxFuture};
+use std::io;
+use std::sync::Arc;
+
+pub struct InformerConfig {
+    pub interface: String,
+    pub port: u16,
+    pub max_connections: usize,
+}
+
+#[derive(Default)]
+pub struct Handler {
+    pub handler: PubSubHandler<Arc<Session>>,
+}
+
+impl Handler {
+    pub fn new(handler: PubSubHandler<Arc<Session>>) -> Self {
+        Self {
+            handler,
+        }
+    }
+    pub fn start_ws(self, config: InformerConfig) -> Result<WsServer, String> {
+        let url = format!("{}:{}", config.interface, config.port);
+        let addr = url.parse().map_err(|_| format!("Invalid WebSockets listen host/port given: {}", url))?;
+        let server = self.handler;
+        let start_result = start_ws(&addr, server, config.max_connections);
+        match start_result {
+            Err(WsError::Io(ref err)) if err.kind() == io::ErrorKind::AddrInUse => {
+                Err(format!("WebSockets address {} is already in use, make sure that another instance of a Codechain node is not running or change the address using the --ws-port options.", addr))
+            },
+            Err(e) => {
+                Err(format!("WebSockets error: {:?}", e))
+            },
+            Ok(server) => {
+                cinfo!(INFORMER, "WebSockets Listening on {}", addr);
+                Ok(server)
+            },
+        }
+    }
+    pub fn event_subscription(&mut self, sender: Sender<Connection>) {
+        self.handler.add_subscription(
+            "Your event subscription begins",
+            ("register", move |params: Params, _, subscriber: Subscriber| {
+                let paramsclone = params.clone();
+                let unwraped_params: Result<Vec<String>, Error> = params.parse();
+                let allparams = unwraped_params.unwrap();
+                println!("####################");
+                println!("{}", allparams[0]);
+                if paramsclone == Params::None {
+                    subscriber
+                        .reject(Error {
+                            code: ErrorCode::ParseError,
+                            message: "Invalid parameters. Subscription rejected.".into(),
+                            data: None,
+                        })
+                        .unwrap();
```

Comment:
Do you know when the `unwrap` fails? Please use `expect` rather than `unwrap` and use a message that contains why you convinced that this will not happen.